SmartDeviceLink/private/SDLSoftButtonManager.m

```diff
@@ -123,16 +123,22 @@ - (void)setSoftButtonObjects:(NSArray<SDLSoftButtonObject *> *)softButtonObjects
     }
 
     // Set the soft button ids. Check to make sure no two soft buttons have the same name, there aren't many soft buttons, so n^2 isn't going to be bad
+    NSUInteger buttonId = 1;
     for (NSUInteger i = 0; i < softButtonObjects.count; i++) {
         NSString *buttonName = softButtonObjects[i].name;
-        softButtonObjects[i].buttonId = i * 100;
+        // HAX: Due to a SYNC 3.0 bug (https://github.com/smartdevicelink/sdl_ios/issues/1793#issue-708356008), a `buttonId` can not be zero. As a workaround we will start the `buttonId`s from 1.
+        if (buttonId == 1001) {
+            buttonId = 1;
+        }
+        softButtonObjects[i].buttonId = buttonId;
```

Comment:
```suggestion
        // HAX: Due to a SYNC 3.0 bug (https://github.com/smartdevicelink/sdl_ios/issues/1793#issue-708356008), a `buttonId` can not be zero. As a workaround we will start the `buttonId`s from 1.
        softButtonObjects[i].buttonId = i + 1;
```

I'm sorry, I misunderstood this code when I told you the other stuff last time. This should be fine.